src/backend/plaf.rs

```diff
@@ -0,0 +1,474 @@
+                                )
+                            });
+                        let offset = (self.offset as i32 + rotation) as usize;
+                        self.plaf_witness.witness[*p_column_index][offset] =
+                            Some(BigUint::from_bytes_le(&value.to_repr()));
+                    }
+                    _ => panic!("selector expression has wrong cPolyExpr enum type"),
+                }
+            }
+
+            self.offset += self.placement.step_height(&cur_step) as usize;
+        }
+    }
+
+    fn assign(&mut self, lhs: Queriable<F>, rhs: F) {
+        if let Some(cur_step) = &self.cur_step {
+            let (p_column_index, rotation) = self.find_plaf_placement(cur_step, lhs);
+
+            let offset = (self.offset as i32 + rotation) as usize;
+            self.plaf_witness.witness[p_column_index][offset] =
+                Some(BigUint::from_bytes_le(&rhs.to_repr()));
+        } else {
+            panic!("jarrl assigning outside a step");
+        }
+    }
+
+    fn find_plaf_placement(
+        &self,
+        step: &StepType<F, StepArgs>,
+        query: Queriable<F>,
+    ) -> (usize, i32) {
+        match query {
+            Queriable::Internal(signal) => self.find_plaf_placement_internal(step, signal),
+
+            Queriable::Forward(forward, next) => {
+                self.find_plaf_placement_forward(step, forward, next)
+            }
+
+            Queriable::Halo2AdviceQuery(_signal, _rotation) => {
+                panic!("Imported Halo2Advice is not supported")
+            }
+
+            _ => panic!("invalid advice assignment on queriable {:?}", query),
+        }
+    }
+
+    fn find_plaf_placement_internal(
+        &self,
+        step: &StepType<F, StepArgs>,
+        signal: InternalSignal,
+    ) -> (usize, i32) {
+        let placement = self.placement.find_internal_signal_placement(step, &signal);
+
+        let p_column_index = self
+            .c_column_id_to_p_column_index
+            .get(&placement.column.uuid())
+            .unwrap_or_else(|| panic!("plaf column not found for internal signal {:?}", signal));
+
+        (*p_column_index, placement.rotation)
+    }
+
+    fn find_plaf_placement_forward(
+        &self,
+        step: &StepType<F, StepArgs>,
+        forward: ForwardSignal,
+        next: bool,
+    ) -> (usize, i32) {
+        let placement = self.placement.get_forward_placement(&forward);
+
+        let super_rotation = placement.rotation
+            + if next {
+                self.placement.step_height(step) as i32
+            } else {
+                0
+            };
+
+        let p_column_index = self
+            .c_column_id_to_p_column_index
+            .get(&placement.column.uuid())
+            .unwrap_or_else(|| panic!("plaf column not found for forward signal {:?}", forward));
+
+        (*p_column_index, super_rotation)
+    }
+}
+
+// This is for debugging only.
+pub fn print_witness(plaf_witness: &pWitness) {
```

Comment:
@qwang98 no code is using this, remove it

Reply:
Fibonacci example is actually using it as a demo.

Reply:
ok, so then move this code to the example. This is not part of the API of the backend, the backend outputs a plaf object, doing something with that plaf object is not part of the backend

Reply:
Fixed